### DISCUSSION TITLE
Decreasing render speed with number of lora's added Flux dev, is this normal?

### DISCUSSION THREAD

**Discussion by alaman64** [lllyasviel/stable-diffusion-webui-forge#2061] [General]
Hi I noticed that my render speed decreases based on the number of lora I add. I'm on Flux Dev, RTX 4090. As you can see the speed drops to about 1/2 with 2 lora's loaded. So 20 second render with 0 lora takes 40 second with 2 lora's. IS this normal? or are there parameters I can change to improve this? Thanks.

flux1-dev-fp8.safetensors, automatic FP16, GPU Weight 21568

0, Lora = 3.16 It/sec
1,  Lora = 1.88
2,  Lora = 1.44
3,  Lora = 1.18

**Comment by FifthDream** [comment]
Oh, i hope someone figures out the power behind that. On my end: 0 loras: 9-13 s/it. One lora: 70 - ∞ s/it. Sigh.

**Reply by alaman64** [reply]
What card are you working with?

**Reply by FifthDream** [reply]
3060 Ti

**Reply by SwedishViking404** [reply]
I have a 4090, and it is usual to see an increase in seconds per iteration as more loras are added as well as a commensurate increase in GPU memory usage per lora. In your case, I would say that it is a memory issue. You could try experimenting with the Swap Method, Swap Location and GPU Weights.

**Comment by iqddd** [comment]
![image](https://github.com/user-attachments/assets/f62ceb7d-358a-44a2-9efa-903304ce6765)
If you can, it's best to use the Automatic method rather than the Automatic (FP16 LORA) method. This one merges Loras with the model. If you have enough RAM for the operation, you'll barely notice any slowdown when using Loras. Plus, it doesn't depend on the number of Loras. Just a heads-up, though: the process can take a while and happens every time you change Loras or their weights.

**Reply by alaman64** [reply]
Thanks, switching to automatic appears to have fixed the issue.
